src/token_contract/src/types/token_utils.nr

```diff
@@ -0,0 +1,219 @@
+mod transferrable;
+mod mintable;
+
+pub comptime fn token(m: Module) -> Quoted {
+    let token_body = transferrable::TRANSFERRABLE_CODE;
+    let mintable_body = mintable::MINTABLE_CODE;
+
+    quote {
+        $token_body
+        $mintable_body
+    }
+}
+
+pub mod TokenLib {
+    use aztec::{
+        context::PrivateContext,
+        note::{note_emission::OuterNoteEmission, note_interface::PartialNote},
+        oracle::random::random,
+        prelude::{AztecAddress, Map, Point, PublicContext, PublicMutable},
+        protocol_types::abis::function_selector::FunctionSelector,
+    };
+
+    use uint_note::uint_note::UintNote;
+
+    use authwit::auth::assert_current_call_valid_authwit;
+
+    use crate::types::balance_set::BalanceSet;
+
+    /** ==========================================================
+     * ======================= LIBRARIES =========================
+     * ======================================================== */
+
+    pub fn _validate_from_private(from: AztecAddress, nonce: Field, context: &mut PrivateContext) {
+        if (!from.eq(context.msg_sender())) {
+            assert_current_call_valid_authwit(context, from);
+        } else {
+            assert(nonce == 0, "invalid nonce");
+        }
+    }
```

Comment:
- [ ] should move to separate lib!
- [ ] add public authwits (comment the call if it doesn't compile)